products/views.py

```diff
@@ -1,3 +1,57 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import Q
+from django.core.paginator import Paginator, EmptyPage
+
+from products.models import *
+
+class MainProductView(View):
+    def get(self, request):
+        try:
+            main_category = request.GET.get('main', None)
+            category      = request.GET.getlist('sub', None)
+            sorting       = request.GET.get('order-by', 'latest')
+            character     = request.GET.getlist('character', None)
+            page          = request.GET.get('page', 1)
```

Comment:
담기는 값이 리스트라면 변수는 복수형으로 선언해주세요.

```suggestion
            main_category = request.GET.get('main', None)
            categories    = request.GET.getlist('sub', None)
            sorting       = request.GET.get('order-by', 'latest')
            characters    = request.GET.getlist('character', None)
            page          = request.GET.get('page', 1)
```